test_scripts/AppServices/OnAppServiceData/010_M2M_invalid_service_id_flow.lua

```diff
@@ -0,0 +1,89 @@
+---------------------------------------------------------------------------------------------------
+--  Precondition: 
+--  1) Application 1 with <appID> is registered on SDL.
+--  2) Application 2 with <appID2> is registered on SDL.
+--  3) Specific permissions are assigned for <appID> with PublishAppService
+--  4) Specific permissions are assigned for <appID2> with OnAppServiceData
```

Comment:
Might be better to say App Service Consumer permissions since it also uses the `GetAppServiceData` RPC